serverless/serverless
# implement version check

## What did you implement:

Closes #2252


## How did you implement it:

I used the same semver package build & used by NPM to check for if the version is satisfied: https://github.com/npm/node-semver
We use an older version 5.0.3 instead of the newest 5.3.0 since the node-https-proxy-agent has some outdated dependencies and hasn't seen a release within a year. Using 5.0.3 satisfied npm shrinkwrap.


## How can we verify it:

1. Create a service and add `frameworkVersion: "=0.5.6"` in the serverless.yaml. It should fail if you want to do `serverless deploy`.
2. Change it to `frameworkVersion: ">=1.0.0"` and it should succeed running `serverless deploy`.


## Todos:

**_Is this ready for review?:**_ YES
